contracts/core/liquidators/TWAPLiquidator.sol

```diff
@@ -0,0 +1,291 @@
+
+        // Validates chunk auction size and chunkAuctionPeriod validity
+        TWAPAuction.validateLiquidatorData(
+            _currentSet,
+            _nextSet,
+            _startingCurrentSetQuantity,
+            twapLiquidatorData
+        );
+
+        // Validates a valid linear auction can be initiated
+        TwoAssetPriceBoundedLinearAuction.validateTwoAssetPriceBoundedAuction(
+            _currentSet,
+            _nextSet
+        );
+
+        // Initiates the TWAP auction
+        TWAPAuction.initializeTWAPAuction(
+            auctions[msg.sender],
+            _currentSet,
+            _nextSet,
+            _startingCurrentSetQuantity,
+            twapLiquidatorData
+        );
+    }
+
+    /**
+     * Reduces the remainingCurrentSet quantity and retrieves the current
+     * bid price for the chunk auction. If this auction completes the chunkAuction,
+     * the lastChunkAuction parameter is updated.
+     * Can only be called by a SetToken during an active auction
+     *
+     * @param _quantity               The currentSetQuantity to rebalance
+     * @return TokenFlow              Struct with array, inflow, and outflow data 
+     */
+    function placeBid(
+        uint256 _quantity
+    )
+        external
+        isValidSet
+        returns (Rebalance.TokenFlow memory)
+    {
+        Auction.validateBidQuantity(auction(msg.sender), _quantity);
+
+        Auction.reduceRemainingCurrentSets(auction(msg.sender), _quantity);
+
+        // If the auction is complete, update the chunk auction end time to the present timestamp
+        if (!hasBiddableQuantity(auction(msg.sender))) {
+            twapAuction(msg.sender).lastChunkAuctionEnd = block.timestamp;    
+        }
+
+        return getBidPrice(msg.sender, _quantity);
+    }
+
+    /**
+     * Retrieves the current auction price for the particular Set
+     *
+     * @param _set                    Address of the SetToken
+     * @param _quantity               The currentSetQuantity to rebalance
+     * @return TokenFlow              Struct with array, inflow, and outflow data
+     */
+    function getBidPrice(
+        address _set,
+        uint256 _quantity
+    )
+        public
+        view
+        returns (Rebalance.TokenFlow memory)
+    {   
+        return LinearAuction.getTokenFlow(chunkAuction(_set), _quantity);
+    }
+
+    /**
+     * Initiates the next chunk auction. Callable by anybody.
+     * @param _set                    Address of the RebalancingSetToken
+     */
+    function iterateChunkAuction(address _set) external {
+        validateNextChunkAuction(twapAuction(_set));
+
+        auctionNextChunk(twapAuction(_set));
+    }
+
+    /**
+     * Validates auction completion and clears auction state. Callable only by a SetToken.
+     */
+    function settleRebalance() external isValidSet {
+        Auction.validateAuctionCompletion(auction(msg.sender));
```

Comment:
This is redundant since `isRebalanceActive` is already checking if there's any biddable amount available in the sum of orderRemaining.add(twapAuction.chunkAuction.auction.remainingCurrentSets). Whereas this is just checking if remainingCurrentSets has a biddable quantity. If this is`false` then `isRebalanceActive` must also be `false`.

Reply:
gotcha